Mathlib/Analysis/InnerProductSpace/Basic.lean

```diff
@@ -905,6 +905,27 @@ protected theorem Complex.inner (w z : ℂ) : ⟪w, z⟫_ℝ = (conj w * z).re :
 
 end RCLikeToReal
 
+/-- An `RCLike` field is a real inner product space. This is not registered as an instance since it
+creates problems with the case `𝕜 = ℝ`, but it can be used in a proof to obtain a real inner product
+space structure on an `RCLike` field.
+
+This is definitionally different from the inner product space structure obtained from
+`InnerProductSpace.rclikeToReal 𝕜 𝕜` because of a non-defeq diamond in the `ℝ`-normed space
+instances:
+`NormedSpace.restrictScalars ℝ 𝕜 𝕜` isn't defeq to
+`NormedAlgebra.toNormedSpace' (𝕜 := ℝ) (𝕜' := 𝕜)` because the former sets
```

Comment:
I think these primed instances are a relic from the no-structure-eta situation in Lean 3. Removing them entirely is out of scope for this PR, but can you use the unprimed one here?